### DISCUSSION TITLE
Install Fails on Big Sur - Linking completions, docs and manpages

### DISCUSSION THREAD

**Discussion by Sam29556** [Homebrew/discussions#600] [Getting started] [answered]
Trying to install on Big Sur, but the install is unsuccessful.  Can someone please assist?

```
Macintosh-3:~ myusername$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> The following new directories will be created:
/usr/local/Cellar
/usr/local/CaskroomPress RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/mkdir -p /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /bin/chmod g+rwx /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/sbin/chown myusername /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/bin/chgrp admin /usr/local/Cellar /usr/local/Caskroom
==> /usr/bin/sudo /usr/sbin/chown myusername /usr/local/Homebrew
==> /usr/bin/sudo /bin/mkdir -p /Users/myusername/Library/Caches/Homebrew
==> /usr/bin/sudo /bin/chmod g+rwx /Users/myusername/Library/Caches/Homebrew
==> /usr/bin/sudo /usr/sbin/chown myusername /Users/myusername/Library/Caches/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at bf448def7 Merge pull request #10364 from SeekingMeaning/version-strict-typecheck
Error: Failed to link all completions, docs and manpages:
Permission denied @ rb_file_s_symlink - (../../../Homebrew/completions/zsh/_brew, /usr/local/share/zsh/site-functions/_brew)
Failed during: /usr/local/bin/brew update --force --quiet
```

**Comment by gromgit** [comment, marked as answer]
What does `ls -lr /usr/local/share/zsh` output?

**Reply by Sam29556** [reply]
total 0
drwxr-xr-x  3 root  wheel  96 Dec 15 09:32 site-functions

**Reply by gromgit** [reply]
Running `sudo chown -R $(whoami): /usr/local/share/zsh` should fix that, then you can run the installer again.

**Reply by Sam29556** [reply]
Success!  Thank you!

**Reply by shameem16** [reply]
Thanks @gromgit !

**Reply by danielgohlke** [reply]
thx

**Reply by pr0b3r7** [reply]
This solution works!

**Reply by codeskipper** [reply]
I ran into this issue on macOS 11.5.1 nearly 10 months later. @gromgit's solution fixed it for me.

**Reply by Ahauk** [reply]
Thanks man, @gromgit's solution fixed it for me as well

**Reply by gondhijagapathi** [reply]
Thanks, dude

**Reply by nincentaur** [reply]
tysm!

**Reply by tumraja** [reply]
Thanks, it saved my day :D

**Reply by ajaxart** [reply]
Thank you @gromgit that worked!

**Reply by sammiekiogora** [reply]
Thank you @gromgit It has worked on macOS Monterey

**Reply by luiseduardosc** [reply]
Thank You, it saved my day ❤️

**Reply by aashishpiitk** [reply]
> Running `sudo chown -R $(whoami): /usr/local/share/zsh` should fix that, then you can run the installer again.

Saved my day man. I had been debugging this for hours

**Reply by phrane** [reply]
I've same issue above (on BigSur) but running chown as shown above doesnt fix for me. 
Permissions still show as ``` drwxr-xr-x  2 root  wheel  64 30 Jun 09:59 site-functions ``` both before and after chown.
Any other ideas welcome. Thanks

**Reply by gromgit** [reply]
> Permissions still show as `drwxr-xr-x 2 root wheel 64 30 Jun 09:59 site-functions` both before and after chown.

1. Run the command exactly as shown: `sudo chown -R $(whoami): /usr/local/share/zsh`. In particular, if you leave out `-R`, it will only chown `/usr/local/share/zsh` itself.
2. Run `ls -ld /usr/local/share/zsh/site-functions` *immediately* after you do the `chown`. If it still shows the directory to be owned by root, disable any anti-virus or other system management software you are running, then do the `chown` and `ls` again.

**Reply by kevinaboy** [reply]
Thanks for the help!

**Reply by Simonasks** [reply]
Thanks. Fixed the issue.
